gpkit/_mosek/mosek_conif.py

```diff
@@ -0,0 +1,246 @@
+    #
+    #       The variable z[0] is special,
+    #       because it's the epigraph of the objective function
+    #       in LSE form. The sign of this variable is not constrained.
+    #
+    #       The variables z[1:] are epigraph terms for "log",
+    #       in constraints that naturally write as
+    #       LSE(Ai @ x + log_ci) <= 0. These variables need to be <= 0.
+    #
+    #       The main constraints on (x, t, z) are described
+    #       in next comment block.
+    #
+    env = mosek.Env()
+    task = env.Task(0, 0)
+    m = A.shape[1]
+    msk_nvars = m + 3 * n_lse + p_lse
+    task.appendvars(msk_nvars)
+    # "x" is free
+    task.putvarboundlist(np.arange(m), [mosek.boundkey.fr] * m,
+                         np.zeros(m), np.zeros(m))
+    # t[3 * i + i] == 1, other components are free.
+    bound_types = [mosek.boundkey.fr, mosek.boundkey.fx, mosek.boundkey.fr]
+    task.putvarboundlist(np.arange(m, m + 3*n_lse), bound_types * n_lse,
+                         np.ones(3*n_lse), np.ones(3*n_lse))
+    # z[0] is free; z[1:] <= 0.
+    bound_types = [mosek.boundkey.fr] + [mosek.boundkey.up] * (p_lse - 1)
+    task.putvarboundlist(np.arange(m + 3*n_lse, msk_nvars), bound_types,
+                         np.zeros(p_lse), np.zeros(p_lse))
+    # t[3*i], t[3*i + 1], t[3*i + 2] belongs to the exponential cone
+    task.appendconesseq([mosek.conetype.pexp] * n_lse, [0.0] * n_lse,
+                        [3] * n_lse, m)
+    #
+    #   Exponential cone affine constraints (other than t[3*i + 1] == 1).
+    #
+    #       For each i in {0, ..., n_lse - 1}, we need
+    #           t[3*i + 2] == A_lse[i, :] @ x + log_c_lse[i] - z[lse_p_idx[i]].
+    #
+    #       For each j from {0, ..., p_lse - 1}, the "t" should also satisfy
+    #           sum(t[3*i] for i where i == lse_p_idx[j]) <= 1.
+    #
+    #       When combined with bound constraints ``t[3*i + 1] == 1``, the
+    #       above constraints imply
+    #           LSE(A_lse[sel, :] @ x + log_c_lse[sel]) <= z[i]
+    #       for ``sel = lse_p_idx == i``.
+    #
+    task.appendcons(n_lse + p_lse)
+    # Linear equations between (x,t,z).
+    #   start with coefficients on "x"
+    rows = list(A_lse.row)
+    cols = list(A_lse.col)
+    vals = list(A_lse.data)
+    #   add coefficients on "t"
+    rows += list(range(n_lse))
+    cols += (m + 3*np.arange(n_lse) + 2).tolist()
+    vals += [-1.0] * n_lse
+    #   add coefficients on "z"
+    rows += list(range(n_lse))
+    cols += [m + 3*n_lse + lse_p_idx[i] for i in range(n_lse)]
+    vals += [-1.0] * n_lse
+    task.putaijlist(rows, cols, vals)
+    cur_con_idx = n_lse
+    # Linear inequalities on certain sums of "t".
+    rows, cols, vals = [], [], []
+    for i in range(p_lse):
+        sels = np.nonzero(lse_p_idx == i)[0]
+        rows.extend([cur_con_idx] * sels.size)
+        cols.extend(m + 3 * sels)
+        vals.extend([1] * sels.size)
+        cur_con_idx += 1
```

Comment:
This step has time complexity ``p_lse * n_lse``. It's possible to pre-compute the ``sels`` selectors in time ``n_lse``. @bqpd -- since you mentioned there's still a mosek 8 vs mosek 9 speed difference.

Reply:
Is the speed difference purely a result of the c versus optimizer interface though @bqpd?

Reply:
I actually noticed a slight speedup on my machine using mosek_conif.

Reply:
@rileyjmurray I profiled it and took out the lines that were causing slowdowns in practice, so now around 94% of the time is spent in the optimize() call on my computer; that's where the speed difference is.

Reply:
@1ozturkbe I saw SPaircraft sped up quite a bit on my computer, but some other codes slowed down considerably

Reply:
makes sense.